src/wizard/analytics-metadata/utils.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import {
+  GeneratedAnalyticsMetadataFragment,
+  getAnalyticsMetadataAttribute,
+} from '@cloudscape-design/component-toolkit/internal/analytics-metadata';
+
+import { WizardProps } from '../interfaces';
+
+export const getNavigationActionDetail = (
+  targetStepIndex: number,
+  reason: WizardProps.NavigationReason,
+  addAction = false,
+  label?: string
+) => {
+  const metadata: GeneratedAnalyticsMetadataFragment = { detail: { targetStepIndex: `${targetStepIndex}`, reason } };
```

Comment:
Only caveat here is to ensure that when processing the data from these events that we remember we have an index and not a step number which would be index+1. In the funnels we normally provided the number and not the index.

Reply:
I try to align this more with the API of the component